src/bin/simhash-find-all.cpp

```diff
@@ -0,0 +1,204 @@
+#include <iostream>
+#include <unordered_set>
+#include <sstream>
+#include <fstream>
+
+#include <getopt.h>
+
+#include "simhash.h"
+
+void usage(int argc, char** argv)
+{
+    std::cout << "usage: " << argv[0]
+              << " --blocks BLOCKS"
+              << " --distance DISTANCE"
+              << " --input INPUT"
+              << " --output OUTPUT"
+              << std::endl
```

Comment:
Martin would say: there's no need to flush after every line. You can just use a literal newline character in your strings to end the lines.